Documentation/spec/docfx_document_schema.md

```diff
@@ -0,0 +1,197 @@
+
+> [!note]
+> As there may be more keywords in JSON schema added to this, it's NOT RECOMMENDED to reuse the keywords already defined in JSON schema here, unless the usage is exactly same.
+
+## 8. Samples
+Here's an sample of the schema. Assume we have the following YAML file:
+```yaml
+title: Web Apps Documentation
+metadata:
+  title: Azure Web Apps Documentation - Tutorials, API Reference
+  meta.description: Learn how to use App Service Web Apps to build and host websites and web applications.
+  services: app-service
+  author: apexprodleads
+  manager: carolz
+  ms.service: app-service
+  ms.tgt_pltfrm: na
+  ms.devlang: na
+  ms.topic: landing-page
+  ms.date: 01/23/2017
+  ms.author: carolz
+sections:
+- title: 5-Minute Quickstarts
+  children:
+  - text: .NET
+    href: app-service-web-get-started-dotnet.md
+  - text: Node.js
+    href: app-service-web-get-started-nodejs.md
+  - text: PHP
+    href: app-service-web-get-started-php.md
+  - text: Java
+    href: app-service-web-get-started-java.md
+  - text: Python
+    href: app-service-web-get-started-python.md
+  - text: HTML
+    href: app-service-web-get-started-html.md
+- title: Step-by-Step Tutorials
+  children:
+  - content: "Create an application using [.NET with Azure SQL DB](app-service-web-tutorial-dotnet-sqldatabase.md) or [Node.js with MongoDB](app-service-web-tutorial-nodejs-mongodb-app.md)"
+  - content: "[Map an existing custom domain to your application](app-service-web-tutorial-custom-domain.md)"
+  - html: "[Bind an existing SSL certificate to your application](app-service-web-tutorial-custom-SSL.md)"
+```
+
+In this sample, we want to use the JSON schema to describe the overall model structure. Further more, the `href` property need to be resolved from the relative path to the final href. The `content` property need to be marked up as a Markdown string. The `metadata` need to be tagged for further custom operations.
+
+Here's the schema to describe these operations:
+
+```json
+{
+    "type": "object",
+    "properties": {
+        "metadata": {
+            "type": "object",
+            "tags": "metadata"
+        },
+        "sections": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "children": {
+                        "type": "array",
+                        "items": {
+                            "type": "object",
+                            "properties": {
+                                "href": {
+                                    "type": "string",
+                                    "behavior": "href"
+                                },
+                                "text": {
+                                    "type": "string"
+                                },
+                                "content": {
+                                    "type": "string",
+                                    "behavior": "markup"
```

Comment:
change to "contentType", use "x-" for other keywords

---
In reply to: [122866576](https://github.com/dotnet/docfx/pull/1771#discussion_r122866576) [](ancestors = 122866576)